lib/dockerregistry/inventory.go

```diff
@@ -1033,6 +1022,22 @@ func (sc *SyncContext) ReadRegistries(
 	sc.ExecRequests(populateRequests, processRequest)
 }
 
+func SplitByKnownRegistries(
+	r RegistryName,
+	rcs []RegistryContext) (RegistryName, ImageName, error) {
+
+	for _, rc := range rcs {
+		//		if inameIdx := strings.LastIndex(string(r), string(rc.Name)); inameIdx != -1 {
+		//			return rc.Name, ImageName(r[inameIdx+1:]), nil
+		//		}
+		if trimmed := strings.TrimPrefix(string(r), string(rc.Name)); len(trimmed) < len(r) {
```

Comment:
I'd probably just do the explicit `strings.HasPrefix` and then a `strings.TrimPrefix`

Reply:
ACK